### DISCUSSION TITLE
Can Immich actually move/reorganize assets?

### DISCUSSION THREAD

**Discussion by mikedolx** [immich-app/immich#123] [Q&A] [answered]
Hi,

I'm looking for a tool, that can move all pictures in my existing library to a new folder structure based on dates. Futhermore, i'd like to add new fotos (from my mobile phone) to that structure?
Does Immich has a feature, where it reorganizes assets into a new folder structure?

**Comment by alextran1502** [comment, marked as answer]
Hi @mikedolx, Immich is used primarily for backing up your photos and videos from your mobile phone, with an interface to view those backed-up assets. It doesn't provide a way to directly reorganize existing assets (if I understand you correctly). However, if what you meant is to have a way to specify your own folder structure when the assets are uploaded to the server, it doesn't have that feature yet. But I've implemented and tested that feature on the server side, I just need to integrate that into the mobile app.

**Reply by mikedolx** [reply]
Hi @alextran1502 ,
yeah i think that goes into the right direction. I basically would like to have the tool to organize my photos in a date based folder structure. I guess i could "upload" my existing library, and then let Immich do the dirty work. I think i'll mark that answer as the solution.

**Reply by mikedolx** [reply]
btw, i don't think i'd need that feature on my mobile phone. But i guess it wouldn't hurt.
